sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,547 @@
+      webResource.headers.set("If-Match", "*");
+    }
+    webResource.headers.set("content-type", "application/atom+xml;type=entry;charset=utf-8");
+    webResource.headers.set("content-length", Buffer.byteLength(webResource.body));
+
+    const atomXmlOperationSpec: AtomXmlOperationSpec = {
+      serializer: serializer,
+      shouldParseResponse: false
+    };
+    webResource.atomXmlOperationSpec = atomXmlOperationSpec;
+
+    return this.sendRequest(webResource);
+  }
+
+  /**
+   * Gets a resource.
+   *
+   * @param {string}             path                                                The resource path.
+   * @param {ResourceSerializer} serializer                                          The XML serializer to use.
+   * @return {Promise<HttpOperationResponse>}
+   */
+  private async _getResource(
+    path: string,
+    serializer: ResourceSerializer
+  ): Promise<HttpOperationResponse> {
+    let webResource: WebResource = new WebResource();
+
+    webResource.method = "GET";
+    webResource.url = this.getUrl(path);
+
+    const atomXmlOperationSpec: AtomXmlOperationSpec = {
+      serializer: serializer,
+      shouldParseResponse: true
+    };
+    webResource.atomXmlOperationSpec = atomXmlOperationSpec;
+
+    return this.sendRequest(webResource);
+  }
+
+  /**
+   * Lists resources.
+   *
+   * @param {string}             path                                                The resource path.
+   * @param {object}             requestBody                                         The resource handler.
+   * @param {ResourceSerializer} serializer                                          The XML serializer to use.
+   * @return {Promise<HttpOperationResponse>}
+   */
+  private async _listResources(
+    path: string,
+    requestBody: any,
+    serializer: ResourceSerializer
+  ): Promise<HttpOperationResponse> {
+    let webResource: WebResource = new WebResource();
+
+    webResource.method = "GET";
+
+    const queryParams: any = {};
+    if (requestBody) {
+      if (requestBody.skip) {
+        queryParams["$skip"] = requestBody.skip;
+      }
+      if (requestBody.top) {
+        queryParams["$top"] = requestBody.top;
+      }
+    }
+
+    webResource.url = this.getUrl(path, queryParams);
+
+    console.log(webResource.url);
+
+    const atomXmlOperationSpec: AtomXmlOperationSpec = {
+      serializer: serializer,
+      shouldParseResponse: true
+    };
+    webResource.atomXmlOperationSpec = atomXmlOperationSpec;
+
+    return this.sendRequest(webResource);
+  }
+
+  /**
+   * Lists resources.
+   *
+   * @param {string}             path                                                The resource path.
+   * @return {Promise<HttpOperationResponse>}
+   */
+  private async _deleteResource(path: any): Promise<HttpOperationResponse> {
+    let webResource: WebResource = new WebResource();
+
+    webResource.method = "DELETE";
+    webResource.url = this.getUrl(path);
+
+    return this.sendRequest(webResource);
```

Comment:
Why is it that we don't need to set the serializer in case of delete? I see the serializer being set for get, create and list scenarios

Reply:
Yes, delete didn't need a serializer, but with the new abstraction we have now pulling out all serialization bits to the serializer this changes now.

Reply:
About the abstraction, earlier I had the serializer take care of just the data serialization bits which was mostly as minimal/necessary. Current abstraction allows SDKs to potentially use their own choice of XML parser utility in place of core-http's built in utilities (xml2js/xmlbuilder). Such as for performant reasons, etc.

But now since we have `deserialize` dealing with the HTTP response and not just the response data, it seems like `serialize()` could construct the HTTP request as well, or is this better left to being done within the `policy` anyway? 
Thinking that since the HTTP req/res representations are core-http specific, the `serializer`s could potentially not have to react to changes to its structure and deal with only the XML data parsing.

Thoughts? @ramya-rao-a @bterlson @chradek

Reply:
Just checking, is this discussion still relevant after what we've discussed offline? (If so, can you rehash it given the changes that have gone in?)

Reply:
As per latest changes, all requests go through serialize & deserialize steps.